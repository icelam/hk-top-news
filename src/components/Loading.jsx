import React from 'react';
import { makeStyles } from '@material-ui/core/styles';

/* Material UI Components */
import CircularProgress from '@material-ui/core/CircularProgress';
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    minHeight: '100vh'
  }
});

const Loading = () => {
  const classes = useStyles();

  return (
    <Grid container direction="column" justify="center" alignItems="center" className={classes.root}>
      <Grid item>
        <CircularProgress spacing={1} className={classes.progress} />
      </Grid>
    </Grid>
  );
};

export default Loading;
