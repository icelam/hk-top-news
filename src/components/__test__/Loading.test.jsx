import React from 'react';
import { render } from '@testing-library/react';
import Loading from '../Loading';

describe('Error Message', () => {
  it('renders correctly', () => {
    const { container, asFragment } = render(<Loading />);
    expect(container.querySelector('.MuiCircularProgress-svg')).toBeInTheDocument();
    expect(container.querySelector('.MuiCircularProgress-circle')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
