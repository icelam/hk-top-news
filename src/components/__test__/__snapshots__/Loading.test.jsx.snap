// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Message renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        spacing="1"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
